diff --git a/dlls/gdi32/bitblt.c b/dlls/gdi32/bitblt.c
index f3b866057eb..d46d7bada49 100644
--- a/dlls/gdi32/bitblt.c
+++ b/dlls/gdi32/bitblt.c
@@ -163,7 +163,7 @@ static BOOL get_vis_rectangles( DC *dc_dst, struct bitblt_coords *dst,
     return intersect_vis_rectangles( dst, src );
 }
 
-void CDECL free_heap_bits( struct gdi_image_bits *bits )
+void free_heap_bits( struct gdi_image_bits *bits )
 {
     HeapFree( GetProcessHeap(), 0, bits->ptr );
 }
diff --git a/dlls/gdi32/dibdrv/dc.c b/dlls/gdi32/dibdrv/dc.c
index 727f3837e33..f818aa02286 100644
--- a/dlls/gdi32/dibdrv/dc.c
+++ b/dlls/gdi32/dibdrv/dc.c
@@ -565,7 +565,7 @@ static inline void unlock_surface( struct windrv_physdev *dev )
     if (GetTickCount() - dev->start_ticks > FLUSH_PERIOD) dev->surface->funcs->flush( dev->surface );
 }
 
-static void CDECL unlock_bits_surface( struct gdi_image_bits *bits )
+static void unlock_bits_surface( struct gdi_image_bits *bits )
 {
     struct window_surface *surface = bits->param;
     surface->funcs->unlock( surface );
diff --git a/dlls/gdi32/gdi_private.h b/dlls/gdi32/gdi_private.h
index 09e432a4d98..01a0145353d 100644
--- a/dlls/gdi32/gdi_private.h
+++ b/dlls/gdi32/gdi_private.h
@@ -622,7 +622,7 @@ static inline void copy_bitmapinfo( BITMAPINFO *dst, const BITMAPINFO *src )
     memcpy( dst, src, get_dib_info_size( src, DIB_RGB_COLORS ));
 }
 
-extern void CDECL free_heap_bits( struct gdi_image_bits *bits ) DECLSPEC_HIDDEN;
+extern void free_heap_bits( struct gdi_image_bits *bits ) DECLSPEC_HIDDEN;
 
 extern HMODULE gdi32_module DECLSPEC_HIDDEN;
 
diff --git a/dlls/wineps.drv/bitmap.c b/dlls/wineps.drv/bitmap.c
index bafc86e9a24..1a98f162702 100644
--- a/dlls/wineps.drv/bitmap.c
+++ b/dlls/wineps.drv/bitmap.c
@@ -148,7 +148,7 @@ static inline DWORD max_ascii85_size(DWORD size)
     return (size + 3) / 4 * 5;
 }
 
-static void CDECL free_heap_bits( struct gdi_image_bits *bits )
+static void free_heap_bits( struct gdi_image_bits *bits )
 {
     HeapFree( GetProcessHeap(), 0, bits->ptr );
 }
diff --git a/dlls/winex11.drv/bitblt.c b/dlls/winex11.drv/bitblt.c
index fad183b0b01..9180fc67013 100644
--- a/dlls/winex11.drv/bitblt.c
+++ b/dlls/winex11.drv/bitblt.c
@@ -914,12 +914,12 @@ BOOL CDECL X11DRV_StretchBlt( PHYSDEV dst_dev, struct bitblt_coords *dst,
 }
 
 
-static void CDECL free_heap_bits( struct gdi_image_bits *bits )
+static void free_heap_bits( struct gdi_image_bits *bits )
 {
     HeapFree( GetProcessHeap(), 0, bits->ptr );
 }
 
-static void CDECL free_ximage_bits( struct gdi_image_bits *bits )
+static void free_ximage_bits( struct gdi_image_bits *bits )
 {
     XFree( bits->ptr );
 }
diff --git a/include/wine/gdi_driver.h b/include/wine/gdi_driver.h
index 124bb41f7d9..bee2b562117 100644
--- a/include/wine/gdi_driver.h
+++ b/include/wine/gdi_driver.h
@@ -54,7 +54,7 @@ struct gdi_image_bits
 {
     void   *ptr;       /* pointer to the bits */
     BOOL    is_copy;   /* whether this is a copy of the bits that can be modified */
-    void  (CDECL *free)(struct gdi_image_bits *);  /* callback for freeing the bits */
+    void  (*free)(struct gdi_image_bits *);  /* callback for freeing the bits */
     void   *param;     /* extra parameter for callback private use */
 };
 
