diff --git a/dlls/dmstyle/style.c b/dlls/dmstyle/style.c
index f8a62604a9..c30189add8 100644
--- a/dlls/dmstyle/style.c
+++ b/dlls/dmstyle/style.c
@@ -228,8 +228,8 @@ static HRESULT WINAPI IDirectMusicStyle8Impl_GetMotif(IDirectMusicStyle8 *iface,
         IDirectMusicSegment **ppSegment)
 {
         IDirectMusicStyle8Impl *This = impl_from_IDirectMusicStyle8(iface);
-        FIXME("(%p, %s, %p): stub\n", This, debugstr_w(pwszName), ppSegment);
-        return S_FALSE;
+	FIXME("(%p, %p, %p): stub\n", This, pwszName, ppSegment);
+	return S_OK;
 }
 
 static HRESULT WINAPI IDirectMusicStyle8Impl_GetDefaultChordMap(IDirectMusicStyle8 *iface,
